gateway/src/main/java/io/camunda/zeebe/gateway/interceptors/impl/AuthenticationInterceptor.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.gateway.interceptors.impl;
+
+import io.camunda.identity.sdk.Identity;
+import io.camunda.identity.sdk.IdentityConfiguration;
+import io.camunda.identity.sdk.authentication.exception.InvalidClaimException;
+import io.camunda.identity.sdk.authentication.exception.InvalidSignatureException;
+import io.camunda.identity.sdk.authentication.exception.JsonWebKeyException;
+import io.camunda.identity.sdk.authentication.exception.TokenDecodeException;
+import io.camunda.identity.sdk.authentication.exception.TokenExpiredException;
+import io.camunda.zeebe.gateway.impl.configuration.AuthenticationCfg;
+import io.camunda.zeebe.gateway.impl.configuration.IdentityCfg;
+import io.grpc.Metadata;
+import io.grpc.ServerCall;
+import io.grpc.ServerCallHandler;
+import io.grpc.ServerInterceptor;
+import io.grpc.Status;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public final class AuthenticationInterceptor implements ServerInterceptor {
```

Comment:
:wrench: This implies it's sort of a "generic" interceptor for authentication, but it really just works for Identity. A more defensive approach would be to either:

1. Make this the `IdentityInterceptor`, and enforce passing only the `Identity` or `IdentityCfg` as params. Then in `Gateway`, you create it only if mode is identity, not if it's not `NONE`.
2. Use a strategy pattern, where based on the mode enum you instantiate the correct, specific interceptor. 

Considering we only have one use case right now, I would lean towards 1.

Reply:
💯  for option 1, will do